File: designpatterns/src/com/renho/behavior/mediator/ConcreteColleagueB.java
package com.renho.behavior.mediator;

public class ConcreteColleagueB extends Colleague {

	public ConcreteColleagueB(Mediator mediator) {
		super(mediator);
	}
	
	public void send(String message) {
		super.getMediator().send(message, this);
	}
	
	public void notify(String message) {
		System.out.println("ConcreteColleagueB get message:" + message);
	}
}

File: ideaworkspace/designpatterns/src/main/scala/com/renho/behavior/mediator/ConcreteColleagueB.scala
package com.renho.behavior.mediator

/**
  * Created by renho on 2016/7/11.
  */
case class ConcreteColleagueB(mediator: Mediator) extends Colleague(mediator: Mediator) {
	def send(message: String): Unit = {
		mediator.send(message, this)
	}
	def notify(message: String): Unit = {
		println("ConcreteColleagueB-notify:" + message)
	}
}
